pkg/materialize/materialized_view.go

```diff
@@ -114,6 +114,7 @@ var materializedViewQuery = NewBaseQuery(`
 		SELECT id, comment
 		FROM mz_internal.mz_comments
 		WHERE object_type = 'materialized-view'
+		AND object_sub_id IS NULL
```

Comment:
This is the actual fix for the duplication issue. The majority of the PR is just testing